{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4dce9619",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import qiime2 as q2\n",
    "\n",
    "import biom\n",
    "import h5py\n",
    "\n",
    "from qiime2.plugins import feature_table, diversity, greengenes2, taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2172354d",
   "metadata": {},
   "source": [
    "## Metadata - General from LIBR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6e76425c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# diagnosis metadata from LIBR \n",
    "libr_md = pd.read_csv('../Updated_Data/DukeCollab_Metadata-9-22-2022.csv', index_col=0)\n",
    "\n",
    "# medications metadata - ATC format \n",
    "medications = pd.read_csv('../Updated_Data/Tulsa_medications_ATC_v1.csv', \n",
    "                         encoding='unicode_escape', index_col=0)\n",
    "medications = medications.loc[medications.atc.notna()]\n",
    "\n",
    "# ATC medications of interest\n",
    "atc_medication_key = ['C10_LipidReducingAgents', 'J01_Antibacterials',\n",
    "                      'N05_Psycholeptics', 'N06_Psychoanaleptics', \n",
    "                      'N05A_Antipsychotics', 'N05B_Anxiolytics', \n",
    "                      'N05C_Hypnotics_and_Sedatives', 'N06A_Antidepressants', \n",
    "                      'N06B_Psychostimulants_for_ADHD_and_Nootropics', \n",
    "                      'N06D_AntiDementia_drugs']\n",
    "\n",
    "for m in atc_medication_key:\n",
    "    code = m.split('_')[0]\n",
    "    users = medications.loc[medications.atc.str.startswith(code)]['id'].unique()\n",
    "    libr_md.insert(libr_md.shape[1], m, np.where(libr_md.index.isin(users), 'True', 'False'))\n",
    "    \n",
    "\n",
    "# exclude anyone who is on antibactials \n",
    "#libr_md = libr_md.loc[libr_md['J01_Antibacterials'] == 'False']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f9b3fe7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "libr_md.to_csv('../Updated_Data/Medications_Data/full_libr_md_with_medications.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d23bb1d",
   "metadata": {},
   "source": [
    "## Metagenomics metadata cleaning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a79ead50",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_dir = '../Updated_Data/T1000Classic_Qiita10424/'\n",
    "\n",
    "# metadata cleaning \n",
    "md_t1000c = q2.Metadata.load(data_dir + '10424_20230201-070656.txt')\n",
    "md_t1000c_df = md_t1000c.to_dataframe()\n",
    "\n",
    "md_t1000c_df['libr_id'] = [np.nan if 'BLANK' in i.upper() else i.split('.')[1].upper() for i in md_t1000c_df.index]\n",
    "md_t1000c_df['cobre'] = np.where((md_t1000c_df['cobre'] != 'Yes') & (md_t1000c_df['libr_id'].notna()), \n",
    "                                 'No', md_t1000c_df['cobre'])\n",
    "md_t1000c_df['cobre'].replace({'not applicable': np.nan, 'not provided': np.nan},\n",
    "                              inplace=True)\n",
    "md_t1000c_df['timepoint'] = np.where(md_t1000c_df.index.str.contains('1year'), \n",
    "                                     '1year', \n",
    "                                     np.where(md_t1000c_df.libr_id.notna(), 'baseline', \n",
    "                                              np.nan))\n",
    "# remove samples that are not in Tulsa 1000 or COBRE or AS484 -> since this in our metadata \n",
    "md_t1000c_filt = md_t1000c_df.loc[(md_t1000c_df['cobre'] == 'Yes') | \n",
    "                                  (md_t1000c_df['tulsa_1000'] == 'yes') | \n",
    "                                  (md_t1000c_df['libr_id'] == 'AS484')]\n",
    "# remove non fecal samples\n",
    "md_t1000c_fecal = md_t1000c_filt.loc[md_t1000c_filt.sample_type == 'feces']\n",
    "\n",
    "# match to samples from LIBR\n",
    "md_t1000c_libr = md_t1000c_fecal.loc[md_t1000c_fecal.libr_id.isin(libr_md.index)]\n",
    "\n",
    "# save for qiime2 \n",
    "md_t1000c_q2 = q2.Metadata(md_t1000c_libr)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa395396",
   "metadata": {},
   "source": [
    "## T1000 Classic Metagenomics Data (qiita ID 10424) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e7ed97f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# MetaG preps to merge \n",
    "metag_preps_paths = {\n",
    "    'prep_12718': data_dir + 'MetaG/156550_ft.qza',\n",
    "    'prep_12719': data_dir + 'MetaG/156547_ft.qza',\n",
    "    'prep_12720': data_dir + 'MetaG/156565_ft.qza',\n",
    "    'prep_12721': data_dir + 'MetaG/156571_ft.qza',\n",
    "    'prep_12722': data_dir + 'MetaG/156579_ft.qza'\n",
    "}\n",
    "\n",
    "metag_preps = [q2.Artifact.load(metag_preps_paths[i]) for i in metag_preps_paths.keys()]\n",
    "merged_wol2 = feature_table.methods.merge(tables=metag_preps, \n",
    "                                          overlap_method='sum').merged_table\n",
    "\n",
    "# 16S preps to merge \n",
    "amp_preps_paths = {\n",
    "    'prep_1428': data_dir + 'Amplicon/131547_feature-table.qza',\n",
    "    'prep_2145': data_dir + 'Amplicon/128379_feature-table.qza',\n",
    "    'prep_2569': data_dir + 'Amplicon/131073_feature-table.qza',\n",
    "    'prep_4004': data_dir + 'Amplicon/128520_feature-table.qza',\n",
    "    'prep_4607': data_dir + 'Amplicon/131180_feature-table.qza',\n",
    "    'prep_4685': data_dir + 'Amplicon/130023_feature-table.qza', \n",
    "    'prep_6173': data_dir + 'Amplicon/132180_feature-table.qza', \n",
    "    'prep_7243': data_dir + 'Amplicon/132690_feature-table.qza', \n",
    "    'prep_8287': data_dir + 'Amplicon/133462_feature-table.qza'\n",
    "}\n",
    "\n",
    "preps_16s = [q2.Artifact.load(amp_preps_paths[i]) for i in amp_preps_paths.keys()]\n",
    "merged_16s = feature_table.methods.merge(tables=preps_16s).merged_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e6c03ce",
   "metadata": {},
   "source": [
    "## Filter to samples only present in LIBR "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "dbb695c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "table_metag = feature_table.methods.filter_samples(table=merged_wol2, \n",
    "                                                   metadata=md_t1000c_q2).filtered_table\n",
    "table_16s = feature_table.methods.filter_samples(table=merged_16s, \n",
    "                                                 metadata=md_t1000c_q2).filtered_table\n",
    "# also filter 16s to greengenes2 \n",
    "gg2_tax = q2.Artifact.load('../Updated_Data/2022.10.taxonomy.asv.nwk.qza')\n",
    "gg2_tree = q2.Artifact.load('../Updated_Data/2022.10.phylogeny.asv.nwk.qza')\n",
    "table_16s = greengenes2.methods.filter_features(feature_table=table_16s, \n",
    "                                                reference=gg2_tree).filtered_feature_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "964b7b54",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before removing duplicates: (962, 17648)\n",
      "After removing duplicates: (670, 14110)\n"
     ]
    }
   ],
   "source": [
    "print('Before removing duplicates: ' + str(table_16s.view(pd.DataFrame).shape))\n",
    "\n",
    "def merge_with_libr(md_df):\n",
    "    out = md_df.reset_index().merge(libr_md.reset_index(), left_on='libr_id', \n",
    "                                    right_on='id')\n",
    "    dropped = out.drop(columns=['id']).set_index('index')\n",
    "    \n",
    "    for c in dropped.columns[dropped.columns.str.contains('DX_')]: \n",
    "        dropped[c] = dropped[c].astype('str')\n",
    "    \n",
    "    return dropped.rename_axis(index='SampleID')\n",
    "\n",
    "# get dataframe representation of tables \n",
    "table_metag_df = table_metag.view(pd.DataFrame)\n",
    "table_16s_df = table_16s.view(pd.DataFrame)\n",
    "\n",
    "# remove duplicates from 16s \n",
    "md_t1000c_libr.insert(loc=md_t1000c_libr.shape[1], column='sequencing_depth', \n",
    "                      value=table_16s_df.sum(axis=1))\n",
    "md_t1000c_libr = md_t1000c_libr.loc[md_t1000c_libr.sequencing_depth.notna()]\n",
    "md_t1000c_sorted = md_t1000c_libr.sort_values('sequencing_depth', ascending=False)\n",
    "md_t1000c_dedup = md_t1000c_sorted.loc[~md_t1000c_sorted.libr_id.duplicated()]\n",
    "md_t1000c_dedup = md_t1000c_dedup.loc[md_t1000c_dedup.sequencing_depth >= 5000]\n",
    "\n",
    "# get separate metadata for metag and 16s \n",
    "metag_md = md_t1000c_libr.reindex(md_t1000c_libr.index.intersection(table_metag_df.index))\n",
    "metag_md.drop(columns=['sequencing_depth'], inplace=True)\n",
    "metag_md.insert(loc=metag_md.shape[1], column='sequencing_depth', \n",
    "                value=table_metag_df.sum(axis=1))\n",
    "metag_md = merge_with_libr(metag_md)\n",
    "metag_md = metag_md.loc[metag_md.sequencing_depth > 500000]\n",
    "md_metag_q2 = q2.Metadata(metag_md)\n",
    "\n",
    "md_16s = md_t1000c_dedup.reindex(md_t1000c_dedup.index.intersection(table_16s_df.index))\n",
    "md_16s = merge_with_libr(md_16s)\n",
    "md_16s_q2 = q2.Metadata(md_16s)\n",
    "\n",
    "libr_16s_filt = feature_table.methods.filter_samples(table=table_16s, \n",
    "                                                     metadata=md_16s_q2).filtered_table\n",
    "libr_metag_filt = feature_table.methods.filter_samples(table=table_metag, \n",
    "                                                       metadata=md_metag_q2).filtered_table\n",
    "print('After removing duplicates: ' + str(libr_16s_filt.view(pd.DataFrame).shape))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f58e13ce",
   "metadata": {},
   "source": [
    "## Match to LIBR IDs and Nightingale samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e0d25fa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# nightingale \n",
    "ng = pd.read_csv('../Updated_Data/Nightingale/Tulsa_Serum_Nightingale_Preprocessed.csv', \n",
    "                 index_col=0).T\n",
    "# reverse log transform \n",
    "ng = 2**ng\n",
    "\n",
    "# diagnosis metadata from LIBR \n",
    "md_t1000c_ng = md_t1000c_libr.loc[md_t1000c_libr['libr_id'].isin(ng.index)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1fe6f3e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the nightingale \n",
    "ng.rename_axis(index='SampleID', inplace=True)\n",
    "ng_cols_to_keep = ng.columns[~(ng.columns.str.contains('pct')) & ~(ng.columns.str.contains('by'))]\n",
    "ng_bt = q2.Artifact.import_data('FeatureTable[Frequency]', ng).view(biom.Table)\n",
    "ng_bt.filter(ng_cols_to_keep, axis='observation')\n",
    "f = h5py.File('../Updated_Data/Microbiome_Nightingale_Integration/nightingale.biom', 'w')\n",
    "ng_bt.to_hdf5(f, 'tulsa')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "39cef6b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_numbers(md): \n",
    "    criteria = {\n",
    "        'ANX': md.loc[md.DX_ANX == 'True'], \n",
    "        'MDD': md.loc[md.DX_MDD == 'True'],\n",
    "        'MDD&ANX': md.loc[(md.DX_MDD == 'True') & (md.DX_ANX == 'True')],\n",
    "        'MDD|ANX': md.loc[(md.DX_MDD == 'True') | (md.DX_ANX == 'True')], \n",
    "        'MDD&~ANX': md.loc[(md.DX_MDD == 'True') & (md.DX_ANX == 'False')], \n",
    "        'ANX&~MDD': md.loc[(md.DX_ANX == 'True') & (md.DX_MDD == 'False')], \n",
    "        'HC': md.loc[md.DX_HC == 'True'], \n",
    "        'Cobre': md.loc[md.cobre == 'Yes'], \n",
    "        'T1000': md.loc[md.cobre == 'No']\n",
    "    }\n",
    "    \n",
    "    out_df = pd.DataFrame(columns=['Number', 'Age', 'Sex (% Female)', 'Medication (% Yes)', 'BMI', \n",
    "                                   'C10_LipidReducingAgents', 'N05_Psycholeptics', 'N05A_Antipsychotics', \n",
    "                                   'N05B_Anxiolytics', 'N05C_Hypnotics_and_Sedatives', 'N06_Psychoanaleptics', \n",
    "                                   'N06A_Antidepressants', 'N06B_Psychostimulants_for_ADHD_and_Nootropics'], \n",
    "                          index=criteria.keys())\n",
    "    \n",
    "    for k in criteria: \n",
    "        out_df['Number'][k] = criteria[k].shape[0]\n",
    "        out_df['Sex (% Female)'][k] = np.round(criteria[k]['Gender'].value_counts(normalize=True)['Female']*100, 2)\n",
    "        out_df['Age'][k] = (str(np.round(criteria[k]['Age'].mean(), 2)) + ' ± ' + \n",
    "                            str(np.round(criteria[k]['Age'].std(), 2))) \n",
    "        out_df['Medication (% Yes)'][k] = np.round(criteria[k]['Medicated'].value_counts(normalize=True)['Medicated']*100, 2)\n",
    "        out_df['BMI'][k] = (str(np.round(criteria[k]['InBody_BMI'].mean(), 2)) + ' ± ' + \n",
    "                            str(np.round(criteria[k]['InBody_BMI'].std(), 2)))\n",
    "        out_df['C10_LipidReducingAgents'][k] = 100-np.round(criteria[k]['C10_LipidReducingAgents'].value_counts(normalize=True)['False']*100, 2)\n",
    "        out_df['N05_Psycholeptics'][k] = 100-np.round(criteria[k]['N05_Psycholeptics'].value_counts(normalize=True)['False']*100, 2)\n",
    "        out_df['N05A_Antipsychotics'][k] = 100-np.round(criteria[k]['N05A_Antipsychotics'].value_counts(normalize=True)['False']*100, 2)\n",
    "        out_df['N05B_Anxiolytics'][k] = 100-np.round(criteria[k]['N05B_Anxiolytics'].value_counts(normalize=True)['False']*100, 2)\n",
    "        out_df['N05C_Hypnotics_and_Sedatives'][k] = 100-np.round(criteria[k]['N05C_Hypnotics_and_Sedatives'].value_counts(normalize=True)['False']*100, 2)\n",
    "        out_df['N06_Psychoanaleptics'][k] = np.round(criteria[k]['N06_Psychoanaleptics'].value_counts(normalize=True)['True']*100, 2)\n",
    "        out_df['N06A_Antidepressants'][k] = np.round(criteria[k]['N06A_Antidepressants'].value_counts(normalize=True)['True']*100, 2)\n",
    "        out_df['N06B_Psychostimulants_for_ADHD_and_Nootropics'][k] = 100-np.round(criteria[k]['N06B_Psychostimulants_for_ADHD_and_Nootropics'].value_counts(normalize=True)['False']*100, 2)\n",
    "        #out_df['N06D_AntiDementia_drugs'] = np.round(criteria[k]['N06D_AntiDementia_drugs'].value_counts(normalize=True)['True']*100, 2)\n",
    "        #out_df['Cobre'][k] = np.round(criteria[k]['cobre'].value_counts(normalize=True)['Yes']*100, 2)\n",
    "    return out_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "06ffcef0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Number</th>\n",
       "      <th>Age</th>\n",
       "      <th>Sex (% Female)</th>\n",
       "      <th>Medication (% Yes)</th>\n",
       "      <th>BMI</th>\n",
       "      <th>C10_LipidReducingAgents</th>\n",
       "      <th>N05_Psycholeptics</th>\n",
       "      <th>N05A_Antipsychotics</th>\n",
       "      <th>N05B_Anxiolytics</th>\n",
       "      <th>N05C_Hypnotics_and_Sedatives</th>\n",
       "      <th>N06_Psychoanaleptics</th>\n",
       "      <th>N06A_Antidepressants</th>\n",
       "      <th>N06B_Psychostimulants_for_ADHD_and_Nootropics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ANX</th>\n",
       "      <td>332</td>\n",
       "      <td>33.41 ± 10.52</td>\n",
       "      <td>77.71</td>\n",
       "      <td>46.99</td>\n",
       "      <td>28.04 ± 5.48</td>\n",
       "      <td>4.52</td>\n",
       "      <td>23.49</td>\n",
       "      <td>2.11</td>\n",
       "      <td>18.98</td>\n",
       "      <td>6.93</td>\n",
       "      <td>41.27</td>\n",
       "      <td>40.36</td>\n",
       "      <td>2.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MDD</th>\n",
       "      <td>471</td>\n",
       "      <td>34.17 ± 11.02</td>\n",
       "      <td>74.52</td>\n",
       "      <td>46.5</td>\n",
       "      <td>28.39 ± 5.55</td>\n",
       "      <td>4.67</td>\n",
       "      <td>19.32</td>\n",
       "      <td>2.34</td>\n",
       "      <td>14.23</td>\n",
       "      <td>5.94</td>\n",
       "      <td>41.61</td>\n",
       "      <td>40.76</td>\n",
       "      <td>3.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MDD&amp;ANX</th>\n",
       "      <td>301</td>\n",
       "      <td>33.35 ± 10.63</td>\n",
       "      <td>77.41</td>\n",
       "      <td>46.51</td>\n",
       "      <td>28.18 ± 5.41</td>\n",
       "      <td>3.32</td>\n",
       "      <td>21.26</td>\n",
       "      <td>2.33</td>\n",
       "      <td>16.94</td>\n",
       "      <td>5.98</td>\n",
       "      <td>42.19</td>\n",
       "      <td>41.2</td>\n",
       "      <td>2.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MDD|ANX</th>\n",
       "      <td>502</td>\n",
       "      <td>34.16 ± 10.92</td>\n",
       "      <td>74.9</td>\n",
       "      <td>46.81</td>\n",
       "      <td>28.28 ± 5.59</td>\n",
       "      <td>5.38</td>\n",
       "      <td>20.92</td>\n",
       "      <td>2.19</td>\n",
       "      <td>15.74</td>\n",
       "      <td>6.57</td>\n",
       "      <td>41.04</td>\n",
       "      <td>40.24</td>\n",
       "      <td>3.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MDD&amp;~ANX</th>\n",
       "      <td>170</td>\n",
       "      <td>35.64 ± 11.56</td>\n",
       "      <td>69.41</td>\n",
       "      <td>46.47</td>\n",
       "      <td>28.75 ± 5.8</td>\n",
       "      <td>7.06</td>\n",
       "      <td>15.88</td>\n",
       "      <td>2.35</td>\n",
       "      <td>9.41</td>\n",
       "      <td>5.88</td>\n",
       "      <td>40.59</td>\n",
       "      <td>40.0</td>\n",
       "      <td>4.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANX&amp;~MDD</th>\n",
       "      <td>31</td>\n",
       "      <td>33.97 ± 9.48</td>\n",
       "      <td>80.65</td>\n",
       "      <td>51.61</td>\n",
       "      <td>26.65 ± 6.04</td>\n",
       "      <td>16.13</td>\n",
       "      <td>45.16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>38.71</td>\n",
       "      <td>16.13</td>\n",
       "      <td>32.26</td>\n",
       "      <td>32.26</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HC</th>\n",
       "      <td>164</td>\n",
       "      <td>30.3 ± 10.72</td>\n",
       "      <td>63.41</td>\n",
       "      <td>7.93</td>\n",
       "      <td>26.71 ± 5.25</td>\n",
       "      <td>4.88</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.05</td>\n",
       "      <td>1.22</td>\n",
       "      <td>1.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cobre</th>\n",
       "      <td>157</td>\n",
       "      <td>29.9 ± 10.13</td>\n",
       "      <td>85.35</td>\n",
       "      <td>17.2</td>\n",
       "      <td>26.69 ± 5.44</td>\n",
       "      <td>2.55</td>\n",
       "      <td>6.37</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.82</td>\n",
       "      <td>3.82</td>\n",
       "      <td>21.66</td>\n",
       "      <td>20.38</td>\n",
       "      <td>3.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1000</th>\n",
       "      <td>513</td>\n",
       "      <td>34.26 ± 11.03</td>\n",
       "      <td>67.84</td>\n",
       "      <td>43.66</td>\n",
       "      <td>28.25 ± 5.52</td>\n",
       "      <td>6.04</td>\n",
       "      <td>18.71</td>\n",
       "      <td>2.14</td>\n",
       "      <td>14.42</td>\n",
       "      <td>5.26</td>\n",
       "      <td>34.89</td>\n",
       "      <td>33.92</td>\n",
       "      <td>2.73</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Number            Age Sex (% Female) Medication (% Yes)  \\\n",
       "ANX         332  33.41 ± 10.52          77.71              46.99   \n",
       "MDD         471  34.17 ± 11.02          74.52               46.5   \n",
       "MDD&ANX     301  33.35 ± 10.63          77.41              46.51   \n",
       "MDD|ANX     502  34.16 ± 10.92           74.9              46.81   \n",
       "MDD&~ANX    170  35.64 ± 11.56          69.41              46.47   \n",
       "ANX&~MDD     31   33.97 ± 9.48          80.65              51.61   \n",
       "HC          164   30.3 ± 10.72          63.41               7.93   \n",
       "Cobre       157   29.9 ± 10.13          85.35               17.2   \n",
       "T1000       513  34.26 ± 11.03          67.84              43.66   \n",
       "\n",
       "                   BMI C10_LipidReducingAgents N05_Psycholeptics  \\\n",
       "ANX       28.04 ± 5.48                    4.52             23.49   \n",
       "MDD       28.39 ± 5.55                    4.67             19.32   \n",
       "MDD&ANX   28.18 ± 5.41                    3.32             21.26   \n",
       "MDD|ANX   28.28 ± 5.59                    5.38             20.92   \n",
       "MDD&~ANX   28.75 ± 5.8                    7.06             15.88   \n",
       "ANX&~MDD  26.65 ± 6.04                   16.13             45.16   \n",
       "HC        26.71 ± 5.25                    4.88               0.0   \n",
       "Cobre     26.69 ± 5.44                    2.55              6.37   \n",
       "T1000     28.25 ± 5.52                    6.04             18.71   \n",
       "\n",
       "         N05A_Antipsychotics N05B_Anxiolytics N05C_Hypnotics_and_Sedatives  \\\n",
       "ANX                     2.11            18.98                         6.93   \n",
       "MDD                     2.34            14.23                         5.94   \n",
       "MDD&ANX                 2.33            16.94                         5.98   \n",
       "MDD|ANX                 2.19            15.74                         6.57   \n",
       "MDD&~ANX                2.35             9.41                         5.88   \n",
       "ANX&~MDD                 0.0            38.71                        16.13   \n",
       "HC                       0.0              0.0                          0.0   \n",
       "Cobre                    0.0             3.82                         3.82   \n",
       "T1000                   2.14            14.42                         5.26   \n",
       "\n",
       "         N06_Psychoanaleptics N06A_Antidepressants  \\\n",
       "ANX                     41.27                40.36   \n",
       "MDD                     41.61                40.76   \n",
       "MDD&ANX                 42.19                 41.2   \n",
       "MDD|ANX                 41.04                40.24   \n",
       "MDD&~ANX                40.59                 40.0   \n",
       "ANX&~MDD                32.26                32.26   \n",
       "HC                       3.05                 1.22   \n",
       "Cobre                   21.66                20.38   \n",
       "T1000                   34.89                33.92   \n",
       "\n",
       "         N06B_Psychostimulants_for_ADHD_and_Nootropics  \n",
       "ANX                                               2.71  \n",
       "MDD                                                3.4  \n",
       "MDD&ANX                                           2.99  \n",
       "MDD|ANX                                           3.19  \n",
       "MDD&~ANX                                          4.12  \n",
       "ANX&~MDD                                           0.0  \n",
       "HC                                                1.83  \n",
       "Cobre                                             3.18  \n",
       "T1000                                             2.73  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_numbers(md_16s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "5fbb1a90",
   "metadata": {},
   "outputs": [],
   "source": [
    "get_numbers(md_16s).to_csv('../Updated_Results/demographic_info_nomeds.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "e86a338a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# do rarefaction as well! \n",
    "libr_metag_rare = feature_table.methods.rarefy(table=libr_metag_filt, \n",
    "                                             sampling_depth=500000).rarefied_table\n",
    "libr_16s_rare = feature_table.methods.rarefy(table=libr_16s_filt,\n",
    "                                             sampling_depth=5000).rarefied_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35d73d4d",
   "metadata": {},
   "source": [
    "## get GG2 taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "2150c5d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_table(table, metadata, group): \n",
    "    md = metadata.to_dataframe()\n",
    "    \n",
    "    criteria = {\n",
    "        'MDD&ANX': md.loc[((md['DX_MDD'] == 'True') & (md['DX_ANX'] == 'True')) | (md['DX_HC'] == 'True')],\n",
    "        'MDD|ANX': md.loc[((md['DX_MDD'] == 'True') | (md['DX_ANX'] == 'True') | (md['DX_HC'] == 'True'))], \n",
    "        'MDD&~ANX': md.loc[((md['DX_MDD'] == 'True') & (md['DX_ANX'] == 'False')) | (md['DX_HC'] == 'True')],\n",
    "        'ANX&~MDD': md.loc[((md['DX_MDD'] == 'False') & (md['DX_ANX'] == 'True')) | (md['DX_HC'] == 'True')],  \n",
    "        'ANX': md.loc[(md['DX_ANX'] == 'True') | (md['DX_HC'] == 'True')], \n",
    "        'MDD': md.loc[(md['DX_MDD'] == 'True') | (md['DX_HC'] == 'True')],\n",
    "        'ANX_MDD_NO_HC': md.loc[((md['DX_MDD'] == 'True') & (md['DX_ANX'] == 'False')) | \n",
    "                                ((md['DX_MDD'] == 'False') & (md['DX_ANX'] == 'True'))]\n",
    "    }\n",
    "    \n",
    "    md_filt = q2.Metadata(criteria[group])\n",
    "    \n",
    "    return feature_table.methods.filter_samples(table=table, metadata=md_filt).filtered_table\n",
    "\n",
    "tables_for_analysis = { \n",
    "    'Amp_MDD_AND_ANX': filter_table(libr_16s_filt, md_16s_q2, 'MDD&ANX'),\n",
    "    'Amp_MDD_OR_ANX': filter_table(libr_16s_filt, md_16s_q2, 'MDD|ANX'),\n",
    "    'Amp_MDD_ONLY': filter_table(libr_16s_filt, md_16s_q2, 'MDD&~ANX'),\n",
    "    'Amp_ANX_ONLY': filter_table(libr_16s_filt, md_16s_q2, 'ANX&~MDD'), \n",
    "    'Amp_ANX_MDD_COMP': filter_table(libr_16s_filt, md_16s_q2, 'ANX_MDD_NO_HC'),\n",
    "    'Amp_MDD': filter_table(libr_16s_filt, md_16s_q2, 'MDD'),\n",
    "    'Amp_ANX': filter_table(libr_16s_filt, md_16s_q2, 'ANX'), \n",
    "    'MetaG_MDD_AND_ANX': filter_table(libr_metag_filt, md_metag_q2, 'MDD&ANX'),\n",
    "    'MetaG_MDD_OR_ANX': filter_table(libr_metag_filt, md_metag_q2, 'MDD|ANX'),\n",
    "    'MetaG_MDD_ONLY': filter_table(libr_metag_filt, md_metag_q2, 'MDD&~ANX'),\n",
    "    'MetaG_ANX_ONLY': filter_table(libr_metag_filt, md_metag_q2, 'ANX&~MDD'), \n",
    "    'MetaG_ANX_MDD_COMP': filter_table(libr_metag_filt, md_metag_q2, 'ANX_MDD_NO_HC'),\n",
    "    'MetaG_MDD': filter_table(libr_metag_filt, md_metag_q2, 'MDD'),\n",
    "    'MetaG_ANX': filter_table(libr_metag_filt, md_metag_q2, 'ANX')\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e081b471",
   "metadata": {},
   "outputs": [],
   "source": [
    "gg2_class_16s = greengenes2.methods.taxonomy_from_table(reference_taxonomy=gg2_tax, \n",
    "                                                        table=tables_for_analysis['Amp_MDD_OR_ANX'])\n",
    "gg2_taxonomy_16s = gg2_class_16s.classification\n",
    "gg2_taxonomy_16s.view(pd.DataFrame).to_csv('../Updated_Data/gg2_taxonomy_16s.csv')\n",
    "\n",
    "gg2_class_metag = greengenes2.methods.taxonomy_from_table(reference_taxonomy=gg2_tax, \n",
    "                                                          table=tables_for_analysis['MetaG_MDD_OR_ANX'])\n",
    "gg2_taxonomy_metag = gg2_class_metag.classification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "7ef06370",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_annotation(x, full_length):\n",
    "    if len(x) == full_length: \n",
    "        return np.nan\n",
    "    else: \n",
    "        split = x.split(';')\n",
    "        start = -1\n",
    "        while len(split[start]) == 3:\n",
    "            start -= 1\n",
    "    return split[start]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9dae7ce8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "168 x 612 <class 'biom.table.Table'> with 102816 nonzero entries (100% dense)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get ng_bt in same format as feature table \n",
    "ng_bt.filter(set(md_16s.libr_id) & set(ng_bt.ids()))\n",
    "libr_to_sampleid = dict(zip(md_16s.libr_id, md_16s.index))\n",
    "ng_bt.update_ids(libr_to_sampleid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "9141ab4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save biom table for uncollapsed \n",
    "for p in tables_for_analysis.keys(): \n",
    "    bt = tables_for_analysis[p].view(biom.Table)\n",
    "    \n",
    "    f = h5py.File('../Updated_Data/RF_Data/LIBR_Data/' + p + '_uncollapsed.biom', 'w')\n",
    "    bt.to_hdf5(f, 'tulsa')\n",
    "    f.close()\n",
    "    \n",
    "    bt = bt.merge(ng_bt, sample='intersection', observation='union') \n",
    "    f = h5py.File('../Updated_Data/RF_Data/LIBR_Nightingale/' + p + '_uncollapsed.biom', 'w')\n",
    "    bt.to_hdf5(f, 'tulsa')\n",
    "    f.close()\n",
    "\n",
    "# collapse the 16S data at the genus level and MetaG data at the species level \n",
    "for t in tables_for_analysis.keys(): \n",
    "    if 'Amp' in t: \n",
    "        collapsed = taxa.methods.collapse(table=tables_for_analysis[t], taxonomy=gg2_taxonomy_16s,\n",
    "                                          level=6).collapsed_table\n",
    "        full_len = 23\n",
    "    elif 'MetaG' in t: \n",
    "        collapsed = taxa.methods.collapse(table=tables_for_analysis[t], taxonomy=gg2_taxonomy_metag,\n",
    "                                          level=7).collapsed_table\n",
    "        full_len = 27\n",
    "    collapsed_df = collapsed.view(pd.DataFrame)\n",
    "    collapsed_df.columns = [get_annotation(i, full_len) for i in collapsed_df.columns]\n",
    "    \n",
    "    if len(collapsed_df.columns[collapsed_df.columns.isna()]) != 0: \n",
    "        collapsed_df.drop(columns=[np.nan], inplace=True)\n",
    "    tables_for_analysis[t] = q2.Artifact.import_data('FeatureTable[Frequency]', collapsed_df)\n",
    "    \n",
    "# save biom table for collapsed \n",
    "for p in tables_for_analysis.keys(): \n",
    "    bt = tables_for_analysis[p].view(biom.Table)\n",
    "    f = h5py.File('../Updated_Data/RF_Data/LIBR_Data/' + p + '_collapsed.biom', 'w')\n",
    "    bt.to_hdf5(f, 'tulsa')\n",
    "    f.close()\n",
    "    \n",
    "    bt = bt.merge(ng_bt, sample='intersection', observation='union') \n",
    "    f = h5py.File('../Updated_Data/RF_Data/LIBR_Nightingale/' + p + '_collapsed.biom', 'w')\n",
    "    bt.to_hdf5(f, 'tulsa')\n",
    "    f.close()\n",
    "\n",
    "metag_md_comp = metag_md.loc[metag_md.index.difference(md_16s.index)]\n",
    "full_metadata = pd.concat([md_16s, metag_md_comp])\n",
    "ng_metadata = md_16s.loc[md_16s.index.intersection(ng_bt.ids())]\n",
    "\n",
    "full_metadata.to_csv('../Updated_Data/RF_Data/LIBR_Data/metadata.tsv', sep='\\t')\n",
    "ng_metadata.to_csv('../Updated_Data/RF_Data/LIBR_Nightingale/metadata.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45cdfd27",
   "metadata": {},
   "source": [
    "## Unmedicated Participants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "5cd7b1a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get table as biom table \n",
    "ft_to_filt = biom.load_table('../Updated_Data/RF_Data/LIBR_Data/Amp_MDD_OR_ANX_uncollapsed.biom')\n",
    "\n",
    "# change ids based on key \n",
    "feat_key_df = pd.read_csv('../Updated_Data/Medications_Data/feature_key.csv', index_col=0)\n",
    "feat_key_dict = dict(zip(feat_key_df['sequences'], feat_key_df.index))\n",
    "ft_to_filt.update_ids(feat_key_dict, axis='observation')\n",
    "\n",
    "# do a prevalence filter of minimum 10 participants \n",
    "prevalence = ft_to_filt.pa(inplace=False).to_dataframe().sum(axis=1)\n",
    "ft_to_filt.filter(prevalence[prevalence > 10].index, axis='observation')\n",
    "\n",
    "# filter metadata to MDD, ANX, or HC only \n",
    "md_filtered = full_metadata.loc[full_metadata.index.intersection(ft_to_filt.ids())]\n",
    "\n",
    "\n",
    "# remove participants that have Yes for C10, N05, or N06\n",
    "md_unmed = md_filtered.loc[(md_filtered.C10_LipidReducingAgents == 'False') & \n",
    "                           (md_filtered.N05_Psycholeptics == 'False') & \n",
    "                           (md_filtered.N06_Psychoanaleptics == 'False')]\n",
    "md_unmed.to_csv('../Updated_Data/Medications_Data/unmedicated_metadata.tsv', sep='\\t')\n",
    "\n",
    "md_unmed_birdman = md_unmed.loc[md_unmed['InBody_BMI'].notna()]\n",
    "md_unmed_birdman.to_csv('../Updated_Data/Medications_Data/unmedicated_metadata_birdman.tsv', sep='\\t')\n",
    "\n",
    "# filter table to unmedicated participants only\n",
    "unmed_table = ft_to_filt.filter(md_unmed_birdman.index, inplace=False)\n",
    "f = h5py.File('../Updated_Data/Medications_Data/unmedicated_table.biom', 'w')\n",
    "unmed_table.to_hdf5(f, 'tulsa')\n",
    "f.close()\n",
    "\n",
    "# fitler table to only participants with anxiety \n",
    "anx_md = md_filtered.loc[(md_filtered['DX_ANX']=='True') & (md_filtered['InBody_BMI'].notna())]\n",
    "anx_md.to_csv('../Updated_Data/Medications_Data/anx_metadata.tsv', sep='\\t')\n",
    "anx_table = ft_to_filt.filter(anx_md.index, inplace=False)\n",
    "f = h5py.File('../Updated_Data/Medications_Data/anx_table.biom', 'w')\n",
    "anx_table.to_hdf5(f, 'tulsa')\n",
    "f.close()\n",
    "\n",
    "# fitler table to only participants with depression \n",
    "mdd_md = md_filtered.loc[(md_filtered['DX_MDD']=='True') & (md_filtered['InBody_BMI'].notna())]\n",
    "mdd_md.to_csv('../Updated_Data/Medications_Data/mdd_metadata.tsv', sep='\\t')\n",
    "mdd_table = ft_to_filt.filter(mdd_md.index, inplace=False)\n",
    "f = h5py.File('../Updated_Data/Medications_Data/mdd_table.biom', 'w')\n",
    "mdd_table.to_hdf5(f, 'tulsa')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "625a48b4",
   "metadata": {},
   "source": [
    "## T1000 - U19 Data (qiita ID 10317 / 13847) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "d2609144",
   "metadata": {},
   "outputs": [],
   "source": [
    "md_13847 = q2.Metadata.load('../../Tulsa_16S/Data/Metadata/13847_20230206-080057.txt')\n",
    "md_t1ku19_df = md_13847.to_dataframe()\n",
    "t1ku19_ids = pd.read_csv('../Data/Metadata/t1000_8.9.21.ids')\n",
    "agp_to_libr = pd.read_csv('../../Tulsa_16S/Data/Metadata/sample_id_to_libr.tsv', sep='\\t',\n",
    "                          index_col='sample_id')['libr_id'].to_dict()\n",
    "\n",
    "md_t1ku19_df['suffix'] = [i.split('.')[-1] for i in md_t1ku19_df.index]\n",
    "md_t1ku19_df['libr_id'] = [agp_to_libr[i] if i in agp_to_libr.keys() else np.nan for i in md_t1ku19_df.suffix]\n",
    "md_t1ku19_df = md_t1ku19_df.loc[(md_t1ku19_df['libr_id'].isin(libr_md['id'].values)) & \n",
    "                                (md_t1ku19_df.sample_type=='feces')]\n",
    "\n",
    "md_13847 = q2.Metadata(md_t1ku19_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "732b0f71",
   "metadata": {},
   "outputs": [],
   "source": [
    "t1ku19_metag = q2.Artifact.load(ft_dir + '159164_feature-table.qza')\n",
    "\n",
    "tables_t1ku19_16s = {\n",
    "    'Prep_11084': q2.Artifact.load(ft_dir + '136087_feature-table.qza'),\n",
    "    'Prep_11095': q2.Artifact.load(ft_dir + '135992_feature-table.qza')\n",
    "}\n",
    "tables_16s = [tables_t1ku19_16s[i] for i in tables_t1ku19_16s.keys()]\n",
    "merged_t1ku19_16s = feature_table.methods.merge(tables=tables_16s).merged_table\n",
    "\n",
    "# filter the tables\n",
    "t1ku19_metag = feature_table.methods.filter_samples(table=t1ku19_metag, \n",
    "                                                    metadata=md_13847).filtered_table\n",
    "t1ku19_16s = feature_table.methods.filter_samples(table=merged_t1ku19_16s, \n",
    "                                                  metadata=md_13847).filtered_table\n",
    "\n",
    "t1ku19_metag = feature_table.methods.filter_features(table=t1ku19_metag, \n",
    "                                                     min_samples=28).filtered_table\n",
    "t1ku19_16s = feature_table.methods.filter_features(table=t1ku19_16s, \n",
    "                                                   min_samples=28).filtered_table\n",
    "\n",
    "t1ku19_metag = feature_table.methods.filter_samples(table=t1ku19_metag, \n",
    "                                                    min_frequency=245000).filtered_table\n",
    "t1ku19_16s = feature_table.methods.filter_samples(table=t1ku19_16s, \n",
    "                                                  min_frequency=5000).filtered_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "bc9a8e36",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples_16s_t1ku19 = t1ku19_16s.view(pd.DataFrame).index\n",
    "samples_metag_t1ku19 = t1ku19_metag.view(pd.DataFrame).index\n",
    "\n",
    "md_16s_t1ku19 = md_t1ku19_df.loc[samples_16s_t1ku19]\n",
    "md_metag_t1ku19 = md_t1ku19_df.loc[samples_metag_t1ku19]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "3558da89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "113"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(md_16s_t1ku19.libr_id.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "7b47b40f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "118"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(md_metag_t1ku19.libr_id.unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87048670",
   "metadata": {},
   "source": [
    "## Combine all samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "2f4d8d8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_16s = pd.concat([filt_md_16s, md_16s_t1ku19])\n",
    "all_metag = pd.concat([filt_md_metag, md_metag_t1ku19])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "ab730bcc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "715"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_16s['libr_id'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "047d6de9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "346"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_metag['libr_id'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "391a3ea6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# bod's metadata \n",
    "all_bod = pd.read_csv('../../U19_Tulsa/Data/Metadata/q2_stool_T1000_U19_sample_metadata.tsv', \n",
    "                      sep='\\t', index_col='sample_name')\n",
    "all_bod_libr = all_bod.loc[all_bod.libr_id.isin(libr_md['id'].values)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "0ce8f420",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/67/8j7qqb5s0cv4wfqrkcwnz9880000gn/T/qiime2/hazedil/data/e8d54267-bef2-42ed-a777-e62cdc658710')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: e8d54267-bef2-42ed-a777-e62cdc658710>"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t1ku19_metag_df = q2.Artifact.load('../../U19_Tulsa/Data/MetaG/clean_feature_table.qza').view(pd.DataFrame)\n",
    "#t1ku19_metag_df.index = ['.'.join(i.split('.')[1:]) for i in t1ku19_metag_df.index]\n",
    "bod_q2 = q2.Artifact.import_data('FeatureTable[Frequency]', \n",
    "                                 t1ku19_metag_df.loc[all_bod_libr.index.intersection(t1ku19_metag_df.index)]) \n",
    "feature_table.visualizers.summarize(table=bod_q2, sample_metadata=q2.Metadata.load('../../U19_Tulsa/Data/Metadata/q2_stool_T1000_U19_sample_metadata.tsv')).visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "4527fee9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>G000005825</th>\n",
       "      <th>G000006605</th>\n",
       "      <th>G000006725</th>\n",
       "      <th>G000006745</th>\n",
       "      <th>G000006785</th>\n",
       "      <th>G000006845</th>\n",
       "      <th>G000006865</th>\n",
       "      <th>G000006925</th>\n",
       "      <th>G000006965</th>\n",
       "      <th>G000007085</th>\n",
       "      <th>...</th>\n",
       "      <th>G900156305</th>\n",
       "      <th>G900156415</th>\n",
       "      <th>G900156525</th>\n",
       "      <th>G900156605</th>\n",
       "      <th>G900156615</th>\n",
       "      <th>G900156675</th>\n",
       "      <th>G900156765</th>\n",
       "      <th>G900156885</th>\n",
       "      <th>G900156915</th>\n",
       "      <th>G900163845</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>000145946</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>154.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>000147488</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>000147497</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>276.0</td>\n",
       "      <td>184.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>000147494</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>000147743</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6906.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X00174066</th>\n",
       "      <td>0.0</td>\n",
       "      <td>15370.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>3514.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X00173305</th>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1119.0</td>\n",
       "      <td>3887.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X00174428</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X00174568</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>639.0</td>\n",
       "      <td>21943.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X00175869</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1167.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>243 rows × 5700 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           G000005825  G000006605  G000006725  G000006745  G000006785  \\\n",
       "000145946         0.0         0.0         0.0         0.0         1.0   \n",
       "000147488         0.0         0.0         0.0         0.0         2.0   \n",
       "000147497         0.0         1.0         0.0         0.0         0.0   \n",
       "000147494         0.0         0.0         0.0         0.0         0.0   \n",
       "000147743         0.0         0.0         0.0         0.0         1.0   \n",
       "...               ...         ...         ...         ...         ...   \n",
       "X00174066         0.0     15370.0         0.0         0.0        53.0   \n",
       "X00173305         0.0         7.0         0.0         1.0         0.0   \n",
       "X00174428         0.0         4.0         0.0         0.0         1.0   \n",
       "X00174568         1.0         0.0         0.0         0.0         3.0   \n",
       "X00175869         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "           G000006845  G000006865  G000006925  G000006965  G000007085  ...  \\\n",
       "000145946         0.0       154.0        91.0         0.0         0.0  ...   \n",
       "000147488         0.0       107.0        21.0         1.0         0.0  ...   \n",
       "000147497         0.0       276.0       184.0         0.0         0.0  ...   \n",
       "000147494         0.0        35.0        22.0         0.0         0.0  ...   \n",
       "000147743         0.0         5.0      6906.0         0.0         0.0  ...   \n",
       "...               ...         ...         ...         ...         ...  ...   \n",
       "X00174066         0.0        31.0      3514.0         0.0         0.0  ...   \n",
       "X00173305         2.0      1119.0      3887.0         0.0         0.0  ...   \n",
       "X00174428         0.0         2.0       182.0         0.0         0.0  ...   \n",
       "X00174568         0.0       639.0     21943.0         0.0         0.0  ...   \n",
       "X00175869         0.0         0.0      1167.0         0.0         0.0  ...   \n",
       "\n",
       "           G900156305  G900156415  G900156525  G900156605  G900156615  \\\n",
       "000145946         0.0         0.0         0.0         0.0         0.0   \n",
       "000147488         0.0         0.0         0.0         0.0         0.0   \n",
       "000147497         0.0         0.0         0.0         0.0         0.0   \n",
       "000147494         0.0         0.0         0.0         0.0         0.0   \n",
       "000147743         0.0         0.0         0.0         0.0         0.0   \n",
       "...               ...         ...         ...         ...         ...   \n",
       "X00174066         0.0         0.0         0.0         0.0         0.0   \n",
       "X00173305         0.0         0.0         0.0         0.0         0.0   \n",
       "X00174428         0.0         0.0         0.0         0.0         0.0   \n",
       "X00174568         0.0         0.0         0.0         0.0         0.0   \n",
       "X00175869         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "           G900156675  G900156765  G900156885  G900156915  G900163845  \n",
       "000145946         0.0         0.0         0.0         0.0         0.0  \n",
       "000147488         0.0         0.0         0.0         0.0         1.0  \n",
       "000147497         0.0         0.0         0.0         0.0         0.0  \n",
       "000147494         0.0         0.0         0.0         0.0         0.0  \n",
       "000147743         0.0         0.0         0.0         0.0         0.0  \n",
       "...               ...         ...         ...         ...         ...  \n",
       "X00174066         0.0         0.0         0.0         0.0         0.0  \n",
       "X00173305         0.0         0.0        11.0         0.0         0.0  \n",
       "X00174428         0.0         0.0         0.0         0.0         0.0  \n",
       "X00174568         0.0         0.0         0.0         0.0         0.0  \n",
       "X00175869         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[243 rows x 5700 columns]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t1ku19_metag_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "ca9239e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>libr_id</th>\n",
       "      <th>visit</th>\n",
       "      <th>Age</th>\n",
       "      <th>Gender</th>\n",
       "      <th>W.HRatio</th>\n",
       "      <th>MedHx_BMI</th>\n",
       "      <th>OASIS</th>\n",
       "      <th>SCOFF</th>\n",
       "      <th>PHQ</th>\n",
       "      <th>PROMIS_AlcoUseStd</th>\n",
       "      <th>...</th>\n",
       "      <th>vitamin_b_supplement_frequency</th>\n",
       "      <th>vitamin_d_supplement_frequency</th>\n",
       "      <th>vivid_dreams</th>\n",
       "      <th>weight_change</th>\n",
       "      <th>weight_kg</th>\n",
       "      <th>whole_eggs</th>\n",
       "      <th>whole_grain_frequency</th>\n",
       "      <th>covid_depressed2</th>\n",
       "      <th>mental_illness2</th>\n",
       "      <th>covid_anxious2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10317.000154892</th>\n",
       "      <td>AA081</td>\n",
       "      <td>V4</td>\n",
       "      <td>43.7</td>\n",
       "      <td>Male</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>32.094595</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.7</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>117.934157</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Not at all</td>\n",
       "      <td>No</td>\n",
       "      <td>Not at all</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000156084</th>\n",
       "      <td>AA115</td>\n",
       "      <td>V3</td>\n",
       "      <td>51.8</td>\n",
       "      <td>Female</td>\n",
       "      <td>0.783784</td>\n",
       "      <td>22.494537</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.6</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>58.513485</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Not at all</td>\n",
       "      <td>No</td>\n",
       "      <td>Not at all</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000150989</th>\n",
       "      <td>AA343</td>\n",
       "      <td>V3</td>\n",
       "      <td>33.8</td>\n",
       "      <td>Female</td>\n",
       "      <td>0.933333</td>\n",
       "      <td>32.277319</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>90.718582</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>One or more days</td>\n",
       "      <td>Yes</td>\n",
       "      <td>More than two days</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000155870</th>\n",
       "      <td>AA576</td>\n",
       "      <td>V4</td>\n",
       "      <td>50.4</td>\n",
       "      <td>Female</td>\n",
       "      <td>0.822222</td>\n",
       "      <td>29.863770</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.7</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>77.564388</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Not at all</td>\n",
       "      <td>No</td>\n",
       "      <td>Not at all</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.X00167095</th>\n",
       "      <td>AA582</td>\n",
       "      <td>V9</td>\n",
       "      <td>51.4</td>\n",
       "      <td>Female</td>\n",
       "      <td>0.826087</td>\n",
       "      <td>32.944822</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Never</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>89.357803</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Not at all</td>\n",
       "      <td>No</td>\n",
       "      <td>Not at all</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000150200</th>\n",
       "      <td>BI206</td>\n",
       "      <td>V2</td>\n",
       "      <td>38.2</td>\n",
       "      <td>Female</td>\n",
       "      <td>0.849057</td>\n",
       "      <td>40.383976</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>2.8</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Increased more than 10 pounds</td>\n",
       "      <td>108.862298</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>One or more days</td>\n",
       "      <td>Yes</td>\n",
       "      <td>More than two days</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000148118</th>\n",
       "      <td>BI365</td>\n",
       "      <td>V2</td>\n",
       "      <td>30.6</td>\n",
       "      <td>Male</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>27.517950</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2.6</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>74.842830</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>One or more days</td>\n",
       "      <td>Yes</td>\n",
       "      <td>More than two days</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000150240</th>\n",
       "      <td>BI380</td>\n",
       "      <td>V3</td>\n",
       "      <td>29.9</td>\n",
       "      <td>Female</td>\n",
       "      <td>0.888889</td>\n",
       "      <td>22.998535</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>2.9</td>\n",
       "      <td>...</td>\n",
       "      <td>Daily</td>\n",
       "      <td>Daily</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>63.503007</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>One or more days</td>\n",
       "      <td>Yes</td>\n",
       "      <td>One or two days</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.X00163898</th>\n",
       "      <td>BI712</td>\n",
       "      <td>V5</td>\n",
       "      <td>26.6</td>\n",
       "      <td>Female</td>\n",
       "      <td>0.878049</td>\n",
       "      <td>30.175598</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>65.770972</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>One or more days</td>\n",
       "      <td>No</td>\n",
       "      <td>More than two days</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.X00164170</th>\n",
       "      <td>BJ578</td>\n",
       "      <td>V9</td>\n",
       "      <td>46.3</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.695133</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Daily</td>\n",
       "      <td>Increased more than 10 pounds</td>\n",
       "      <td>86.182653</td>\n",
       "      <td>Daily</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>One or more days</td>\n",
       "      <td>No</td>\n",
       "      <td>More than two days</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>247 rows × 231 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                libr_id visit   Age  Gender  W.HRatio  MedHx_BMI  OASIS  \\\n",
       "sample_name                                                               \n",
       "10317.000154892   AA081    V4  43.7    Male  0.909091  32.094595      0   \n",
       "10317.000156084   AA115    V3  51.8  Female  0.783784  22.494537      0   \n",
       "10317.000150989   AA343    V3  33.8  Female  0.933333  32.277319     14   \n",
       "10317.000155870   AA576    V4  50.4  Female  0.822222  29.863770      0   \n",
       "10317.X00167095   AA582    V9  51.4  Female  0.826087  32.944822      0   \n",
       "...                 ...   ...   ...     ...       ...        ...    ...   \n",
       "10317.000150200   BI206    V2  38.2  Female  0.849057  40.383976     13   \n",
       "10317.000148118   BI365    V2  30.6    Male  0.900000  27.517950     11   \n",
       "10317.000150240   BI380    V3  29.9  Female  0.888889  22.998535      9   \n",
       "10317.X00163898   BI712    V5  26.6  Female  0.878049  30.175598     10   \n",
       "10317.X00164170   BJ578    V9  46.3  Female       NaN  29.695133     13   \n",
       "\n",
       "                 SCOFF  PHQ  PROMIS_AlcoUseStd  ...  \\\n",
       "sample_name                                     ...   \n",
       "10317.000154892      0    0                3.7  ...   \n",
       "10317.000156084      0    0                4.6  ...   \n",
       "10317.000150989      1   11                NaN  ...   \n",
       "10317.000155870      0    0                5.7  ...   \n",
       "10317.X00167095      0    0                NaN  ...   \n",
       "...                ...  ...                ...  ...   \n",
       "10317.000150200      2   13                2.8  ...   \n",
       "10317.000148118      0   10                2.6  ...   \n",
       "10317.000150240      0   11                2.9  ...   \n",
       "10317.X00163898      2   10                NaN  ...   \n",
       "10317.X00164170      1   17                2.0  ...   \n",
       "\n",
       "                 vitamin_b_supplement_frequency  \\\n",
       "sample_name                                       \n",
       "10317.000154892                           Never   \n",
       "10317.000156084                           Never   \n",
       "10317.000150989                           Never   \n",
       "10317.000155870                           Never   \n",
       "10317.X00167095      Rarely (a few times/month)   \n",
       "...                                         ...   \n",
       "10317.000150200                           Never   \n",
       "10317.000148118                           Never   \n",
       "10317.000150240                           Daily   \n",
       "10317.X00163898                           Never   \n",
       "10317.X00164170                           Never   \n",
       "\n",
       "                 vitamin_d_supplement_frequency  \\\n",
       "sample_name                                       \n",
       "10317.000154892                           Never   \n",
       "10317.000156084                           Never   \n",
       "10317.000150989                           Never   \n",
       "10317.000155870                           Never   \n",
       "10317.X00167095      Regularly (3-5 times/week)   \n",
       "...                                         ...   \n",
       "10317.000150200                           Never   \n",
       "10317.000148118                           Never   \n",
       "10317.000150240                           Daily   \n",
       "10317.X00163898                             NaN   \n",
       "10317.X00164170                           Never   \n",
       "\n",
       "                                  vivid_dreams                  weight_change  \\\n",
       "sample_name                                                                     \n",
       "10317.000154892                          Never                Remained stable   \n",
       "10317.000156084                          Never                Remained stable   \n",
       "10317.000150989     Rarely (a few times/month)                Remained stable   \n",
       "10317.000155870     Rarely (a few times/month)                Remained stable   \n",
       "10317.X00167095                          Never                Remained stable   \n",
       "...                                        ...                            ...   \n",
       "10317.000150200                          Never  Increased more than 10 pounds   \n",
       "10317.000148118  Occasionally (1-2 times/week)                Remained stable   \n",
       "10317.000150240  Occasionally (1-2 times/week)                Remained stable   \n",
       "10317.X00163898     Rarely (a few times/month)                Remained stable   \n",
       "10317.X00164170                          Daily  Increased more than 10 pounds   \n",
       "\n",
       "                  weight_kg                     whole_eggs  \\\n",
       "sample_name                                                  \n",
       "10317.000154892  117.934157     Regularly (3-5 times/week)   \n",
       "10317.000156084   58.513485  Occasionally (1-2 times/week)   \n",
       "10317.000150989   90.718582  Occasionally (1-2 times/week)   \n",
       "10317.000155870   77.564388   Rarely (less than once/week)   \n",
       "10317.X00167095   89.357803   Rarely (less than once/week)   \n",
       "...                     ...                            ...   \n",
       "10317.000150200  108.862298     Regularly (3-5 times/week)   \n",
       "10317.000148118   74.842830  Occasionally (1-2 times/week)   \n",
       "10317.000150240   63.503007     Regularly (3-5 times/week)   \n",
       "10317.X00163898   65.770972  Occasionally (1-2 times/week)   \n",
       "10317.X00164170   86.182653                          Daily   \n",
       "\n",
       "                         whole_grain_frequency  covid_depressed2  \\\n",
       "sample_name                                                        \n",
       "10317.000154892  Occasionally (1-2 times/week)        Not at all   \n",
       "10317.000156084   Rarely (less than once/week)        Not at all   \n",
       "10317.000150989     Regularly (3-5 times/week)  One or more days   \n",
       "10317.000155870     Regularly (3-5 times/week)        Not at all   \n",
       "10317.X00167095     Regularly (3-5 times/week)        Not at all   \n",
       "...                                        ...               ...   \n",
       "10317.000150200     Regularly (3-5 times/week)  One or more days   \n",
       "10317.000148118  Occasionally (1-2 times/week)  One or more days   \n",
       "10317.000150240   Rarely (less than once/week)  One or more days   \n",
       "10317.X00163898     Regularly (3-5 times/week)  One or more days   \n",
       "10317.X00164170     Regularly (3-5 times/week)  One or more days   \n",
       "\n",
       "                 mental_illness2      covid_anxious2  \n",
       "sample_name                                           \n",
       "10317.000154892               No          Not at all  \n",
       "10317.000156084               No          Not at all  \n",
       "10317.000150989              Yes  More than two days  \n",
       "10317.000155870               No          Not at all  \n",
       "10317.X00167095               No          Not at all  \n",
       "...                          ...                 ...  \n",
       "10317.000150200              Yes  More than two days  \n",
       "10317.000148118              Yes  More than two days  \n",
       "10317.000150240              Yes     One or two days  \n",
       "10317.X00163898               No  More than two days  \n",
       "10317.X00164170               No  More than two days  \n",
       "\n",
       "[247 rows x 231 columns]"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_bod"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
